wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/provisioncredential/EvidenceRequest.kt

```diff
@@ -598,4 +606,156 @@ fun <ResultT> EvidenceRequestIcaoView(
             }
         }
     }
-}
\ No newline at end of file
+}
+
+@Composable
+fun EvidenceRequestSelfieVideoView(
+    evidenceRequest: EvidenceRequestSelfieVideo,
+    provisioningViewModel: ProvisioningViewModel,
+    permissionTracker: PermissionTracker,
+    walletServerProvider: WalletServerProvider,
+    documentStore: DocumentStore
+) {
+    if (evidenceRequest.poseSequence.isEmpty()) {
+        throw IllegalArgumentException("Pose sequence must not be empty")
+    }
+
+    var stage by remember { mutableIntStateOf(-1) }
+    var buttonEnabled by remember { mutableStateOf(false) }
+    var errorMessage by remember { mutableStateOf("") }
+    val context = LocalContext.current
+    val activity = context.getActivity()!!
+    val selfieRecorder: SelfieRecorder = remember {
+        SelfieRecorder(
+            activity,
+            onRecordingStarted = {
+                buttonEnabled = true
+            },
+            onFinished = { selfieResult: ByteArray ->
+                Logger.i(TAG, "Selfie video finished.")
+                if (selfieResult.isEmpty()) {
```

Comment:
Yeah, I'm not thrilled with the UX here. There's no indication why it failed (hitting the size limit is one case). It may be better to throw an exception and handle the different exception cases in here, with different messages. I'll add a TODO for now, since some of this may be changing as I work on the OOM issues.